test/test_sparse_csr.py

```diff
@@ -2394,6 +2394,105 @@ def test_sampled_addmm_errors(self, device, dtype):
         with self.assertRaisesRegex(RuntimeError, r"Expected mat2 to have strided layout"):
             torch.sparse.sampled_addmm(a_sparse, a, a_sparse)
 
+    @onlyCPU
+    @dtypes(torch.float32, torch.float64, torch.bfloat16)
+    def test_sparse_mm_reduce_sum(self, device, dtype):
+        def run_test(m, n, k, nnz, train, sparse_csc):
+            if sparse_csc:
+                sparse = self.genSparseCSCTensor((k, m), nnz, dtype=dtype, device=device, index_dtype=torch.int64)
+            else:
+                sparse = self.genSparseCSRTensor((m, k), nnz, dtype=dtype, device=device, index_dtype=torch.int64)
+
+            mat = torch.randn(k, n, dtype=dtype)
+            dense = spase.to_dense()
+            ref_mat = mat.clone()
+
+            if train:
+                sparse.requires_grad_()
+                mat.requires_grad_()
+                dense.requires_grad_()
+                ref_mat.requires_grad_()
+
+            out = torch.sparse.mm(csr, mat, 'sum')
```

Comment:
@mingfeima I don't understand how this code runs. `csr` has not been previously defined in this method?

Reply:
Oh sorry, this is my mistake! the indentation for `run_test` is wrong! Fixed!